lib/ansible/template/safe_eval.py

```diff
@@ -139,11 +140,11 @@ def generic_visit(self, node, inside_call=False):
     try:
         parsed_tree = ast.parse(expr, mode='eval')
         cnv.visit(parsed_tree)
-        compiled = compile(parsed_tree, expr, 'eval')
+        compiled = compile(parsed_tree, to_native(expr), 'eval')
         # Note: passing our own globals and locals here constrains what
         # callables (and other identifiers) are recognized.  this is in
         # addition to the filtering of builtins done in CleansingNodeVisitor
-        result = eval(compiled, OUR_GLOBALS, dict(locals))
+        result = wrap_to_text(eval(compiled, OUR_GLOBALS, dict(locals)))
 
```

Comment:
This looks like a clean way to do this, however, recursively changing the types may be slow.  If i understand why this is needed correctly, we probably could do this only in python2 and about paying the cost on python3.

We should also use the errors argument in these (surrogate_or_strict) so that we get an error with bad bytes instead of omitting those from the output

Reply:
:+1: